connection.go

```diff
@@ -28,82 +21,48 @@ type DatabendConn struct {
 	cfg         *Config
 	cancel      context.CancelFunc
 	closed      int32
-	stmts       []*databendStmt
 	logger      *log.Logger
 	rest        *APIClient
 	batchMode   bool
 	batchInsert func() error
 }
 
 func (dc *DatabendConn) exec(ctx context.Context, query string, args ...driver.Value) (driver.Result, error) {
-	respCh := make(chan QueryResponse)
-	errCh := make(chan error)
-	dc.rest.NextQuery()
-
-	go func() {
-		err := dc.rest.QuerySync(ctx, query, args, respCh)
-		errCh <- err
-	}()
-
-	for {
-		select {
-		case err := <-errCh:
-			if err != nil {
-				return emptyResult, err
-			} else {
-				return emptyResult, nil
-			}
-		case resp := <-respCh:
-			b, err := json.Marshal(resp.Data)
-			if err != nil {
-				return emptyResult, err
-			}
-			_, _ = io.Copy(io.Discard, bytes.NewReader(b))
-		}
+	_, err := dc.rest.QuerySync(ctx, query, args)
+	if err != nil {
+		return emptyResult, err
 	}
+	return emptyResult, nil
 }
 
-func (dc *DatabendConn) query(ctx context.Context, query string, args ...driver.Value) (driver.Rows, error) {
-	var r0 *QueryResponse
-	dc.rest.NextQuery()
-
-	err := retry.Do(
-		func() error {
-			r, err := dc.rest.DoQuery(ctx, query, args)
-			if err != nil {
-				return err
-			}
-			r0 = r
-			return nil
-		},
-		// other err no need to retry
-		retry.RetryIf(func(err error) bool {
-			if err != nil && (IsProxyErr(err) || strings.Contains(err.Error(), ProvisionWarehouseTimeout)) {
-				return true
-			}
-			return false
-		}),
-		retry.Delay(2*time.Second),
-		retry.Attempts(5),
-		retry.DelayType(retry.FixedDelay),
-	)
+func (dc *DatabendConn) query(ctx context.Context, query string, args ...driver.Value) (rows driver.Rows, err error) {
```

Comment:
Do not remove the retry logic in `query`.

Reply:
retry is moved to seperate requests